_blog/2023-12-20-loco-rust-rails.mdx

```diff
@@ -0,0 +1,219 @@
+    // do something with context (database, etc)
+    format::text("hello")
+}
+
+pub fn routes() -> Routes {
+    Routes::new()
+        .prefix("test")
+        .add("/", get(hello))
+        .add("/echo", post(echo))
+}
+```
+Now you can add any routes you want to this file and it will be put under this controller when it's added to the `routes()` function in the file. Route-wise, this means it'll be all under the same route. You can access the database connection from the provided `State` - unlike in Axum normally, you don't need to create this yourself.
+
+The great thing about Loco's routing is that everything you know from using Axum can be applied here - so if you know how to write your own extractors, write middleware, and other things this can all be used in Loco since it essentially builds on top of Axum. 
+
+Once you've finished adding all the controllers and routes you want, you can use `cargo run routes` to display all of the routes your application currently has.
+
+## Migrations
+Handling migrations in Loco is quite easy through the CLI - to generate a migration you can just use `cargo run generate migration test`, which generates a migration for the `test` table. The files generated can be found in `migrations` (in the project root) and use `sea_orm`, which is an ORM that builds off of `sqlx` - the most, one of the most popular async-based SQL libraries.
+
+When you generate your migrations and go to the file itself, it should look something like this:
+```rust
+use std::borrow::BorrowMut;
+
+use loco_rs::schema::*;
+use sea_orm_migration::prelude::*;
+
+#[derive(DeriveMigrationName)]
+pub struct Migration;
+
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                table_auto(Tests::Table)
+                    .col(pk_auto(Tests::Id).borrow_mut())
+                    .to_owned(),
+            )
+            .await
+    }
+
+    async fn down(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .drop_table(Table::drop().table(Tests::Table).to_owned())
+            .await
+    }
+}
+
+#[derive(DeriveIden)]
+enum Tests {
+    Table,
+    Id,
+
+}
+```
+From this file, you can add more columns as required - make sure to reflect this in the enum at the bottom! If you are stuck on how to expand your migrations, you can always check the base migration files or you can check out the [`sea-orm` documentation.](https://www.sea-ql.org/SeaORM/docs/index/)
+
+When you're done adding migrations, all you need to do is use `cargo run db migrate` and the CLI will do all of the heavy lifting for you - nothing else is needed!
+
+## Models
+Now that you've generated a table migration, you probably want to generate a model that your application can use. Once you've generated the migration, you'll want to run the following:
+```bash
+cargo run generate model <name-of-migration>
```

Comment:
probably better to say `name of model`, and then you have the option of generating a full model with 

```
cargo loco generate model movies title:string rating:int
```

this saves a lot of work, as you get a full blown model ready to use